gcloud-java-compute/src/main/java/com/google/gcloud/compute/Snapshot.java

```diff
@@ -0,0 +1,208 @@
+
+    private final Compute compute;
+    private final SnapshotInfo.BuilderImpl infoBuilder;
+
+    Builder(Compute compute, SnapshotId snapshotId, DiskId sourceDisk) {
+      this.compute = compute;
+      this.infoBuilder = new SnapshotInfo.BuilderImpl();
+      this.infoBuilder.snapshotId(snapshotId);
+      this.infoBuilder.sourceDisk(sourceDisk);
+    }
+
+    Builder(Snapshot snapshot) {
+      this.compute = snapshot.compute;
+      this.infoBuilder = new SnapshotInfo.BuilderImpl(snapshot);
+    }
+
+    @Override
+    Builder id(String id) {
+      infoBuilder.id(id);
+      return this;
+    }
+
+    @Override
+    Builder creationTimestamp(Long creationTimestamp) {
+      infoBuilder.creationTimestamp(creationTimestamp);
+      return this;
+    }
+
+    @Override
+    public Builder snapshotId(SnapshotId snapshotId) {
+      infoBuilder.snapshotId(snapshotId);
+      return this;
+    }
+
+    @Override
+    public Builder description(String description) {
+      infoBuilder.description(description);
+      return this;
+    }
+
+    @Override
+    Builder status(Status status) {
+      infoBuilder.status(status);
+      return this;
+    }
+
+    @Override
+    Builder diskSizeGb(Long diskSizeGb) {
+      infoBuilder.diskSizeGb(diskSizeGb);
+      return this;
+    }
+
+    @Override
+    Builder licenses(List<LicenseId> licenses) {
+      infoBuilder.licenses(licenses);
+      return this;
+    }
+
+    @Override
+    public Builder sourceDisk(DiskId sourceDisk) {
+      infoBuilder.sourceDisk(sourceDisk);
+      return this;
+    }
+
+    @Override
+    Builder sourceDiskId(String sourceDiskId) {
+      infoBuilder.sourceDiskId(sourceDiskId);
+      return this;
+    }
+
+    @Override
+    Builder storageBytes(Long storageBytes) {
+      infoBuilder.storageBytes(storageBytes);
+      return this;
+    }
+
+    @Override
+    Builder storageBytesStatus(StorageBytesStatus storageBytesStatus) {
+      infoBuilder.storageBytesStatus(storageBytesStatus);
+      return this;
+    }
+
+    @Override
+    public Snapshot build() {
+      return new Snapshot(compute, infoBuilder);
+    }
+  }
+
+  Snapshot(Compute compute, SnapshotInfo.BuilderImpl infoBuilder) {
+    super(infoBuilder);
+    this.compute = checkNotNull(compute);
+    this.options = compute.options();
+  }
+
+  /**
+   * Checks if this snapshot exists.
+   *
+   * @return {@code true} if this snapshot exists, {@code false} otherwise
+   * @throws ComputeException upon failure
+   */
+  public boolean exists() {
+    return reload(Compute.SnapshotOption.fields()) != null;
+  }
+
+  /**
+   * Fetches current snapshot's latest information. Returns {@code null} if the snapshot does not
+   * exist.
+   *
+   * @param options snapshot options
+   * @return a {@code Snapshot} object with latest information or {@code null} if not found
+   * @throws ComputeException upon failure
+   */
+  public Snapshot reload(Compute.SnapshotOption... options) {
```

Comment:
Nit: not a big deal, but could import SnapshotOption and OperationOption and remove "Compute." if you want